### DISCUSSION TITLE
Import mysql database from Local Computer

### DISCUSSION THREAD

**Discussion by Bishalsingh** [planetscale/discussion#201] [Documentation] [answered]
I am able to successfully log in to my cpanel server where my database is located and dump them in my local computer. 

The issue arises with import database function in planetscale. No matter what I do, it won't connect to the database. 

I think there must be something in the pscale shell by which we can import database from local computer.

**Comment by fooforge** [comment, marked as answer]
Hey @Bishalsingh,

Apologies for the slow response! This seems to have slipped through the cracks.

If you have a local dump, the easiest method to get your data imported to PlanetScale is through our [CLI tool `pscale`](https://planetscale.com/cli).

In the below example, I'm opening a proxy connection through `pscale connect` to then connect to my PlanetScale database via the mysql client. Also note that I have used `&` to run `pscale connect` in the background so that I could carry on from within the same terminal session. Later I have used `fg %1` to bring the process back to the front and hit `CTRL+C` to terminate it.

```​
✦ ❯ pscale connect DATABASE BRANCH &
[1] 63208
​
✦2 ❯ Tried address 127.0.0.1:3306, but it's already in use. Picking up a random port ...
Secure connection to database DATABASE and branch BRANCH is established!.
​
Local address to connect your application: 127.0.0.1:62222 (press ctrl-c to quit)
​
✦2 ❯ mysql -u root -h 127.0.0.1 -P 62222
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 9877
Server version: 8.0.23-vitess
​
Copyright (c) 2000, 2022, Oracle and/or its affiliates.
​
Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.
​
Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.
​
mysql> ^DBye
​
✦2 ❯ fg %1
pscale connect DATABASE BRANCH
^C
✦ ❯
```

All you would likely need to add is an input redirect with your dump file, like so: `mysql -u root -h 127.0.0.1 -P 62222 < dump.sql`


Cheers,
Mike

**Reply by sbbu** [reply]
Is there any way to do this anymore? I get the following error trying to pipe anything into pscale shell
`Error: pscale shell only works in interactive mode`

**Reply by jgreet** [reply]
The original example is using `mysql` with `pscale connect` and still works:

```sh
~$ pscale connect tla test
Secure connection to database tla and branch test is established!.

Local address to connect your application: 127.0.0.1:3306 (press ctrl-c to quit)
```

```sh
mysql -h 127.0.0.1 -P 3306 < dump.sql
```

Inside of a `pscale shell` session you could use `source dump.sql` to read in the file.

https://dev.mysql.com/doc/refman/8.0/en/mysql-batch-commands.html
